### PR TITLE
Allow variable amounts for donations (bug fix)

### PR DESCRIPTION
**Pull request by kumar303**
This wasn't checking configured product amounts correctly.

### CODE REVIEW

File: lib/brains/forms.py [hunk 1]
```diff
@@ -249,7 +249,7 @@ def clean(self):
                 code='invalid')
 
         amount = self.cleaned_data.get('amount')
-        if (amount and product.amount != amount):
+        if (product.amount and product.amount != amount):
```
Comment (andymckay) [inline lib/brains/forms.py]: you just changed it back to what I had the first time :)

**Comment by andymckay** [comment]
r+ because it means i was right the first time.

File: lib/brains/forms.py (hunk 1 above)
Comment (kumar303) [inline lib/brains/forms.py]: well why did you change it? :fried_shrimp:

Comment (andymckay) [inline lib/brains/forms.py]: So confused :confetti_ball: :whale2:
